File: 1701. Average Waiting Time/1701. Average Waiting Time.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        currentTime = 0
        totalwaitTime = 0
        
        for customer in customers:
            arrival, time = customer
            
            if currentTime < arrival:
                currentTime = arrival
                
            waitTime = currentTime + time - arrival
            totalwaitTime += waitTime
            
            currentTime += time
        
        return totalwaitTime / len(customers)

File: 1701. Average Waiting Time/1701. Average Waiting Time.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        double res=1;
        int start=customers[0][0];
        int end=start+customers[0][1];;
        double totalTime=end-customers[0][0];
        for(int i=1;i<customers.size();i++)
        {
             
            
            if(end>=customers[i][0])
            {
                start=end;
            }
            else
            {
                start=customers[i][0];
            }
           
            end=start+customers[i][1];
            totalTime+=end-customers[i][0];

        }
        return totalTime/customers.size();
        
    }
};